chat-plugins/roomsettings.js

```diff
@@ -59,7 +59,10 @@ class RoomSettings {
 		return modchatOutput.join(' ');
 	}
 	modjoin() {
-		if (!this.user.can('makeroom') && !this.room.isPersonal) return this.button(this.room.modjoin ? this.room.modjoin : 'off', true);
+		if (!this.user.can('makeroom') && !this.room.isPersonal ||
+			!this.user.can('editroom', null, this.room)) {
```

Comment:
My intent there was to clarify the precedence without drowing everything in a sea of parens, because I keep forgetting if && or || has a higher precedence :(
I saw them being used a bunch so I figured it wouldn't be too bad style-wise.